modules/core/include/opencv2/core/ocl.hpp

```diff
@@ -290,6 +294,21 @@ class CV_EXPORTS Context
 
     void release();
 
+    class CV_EXPORTS UserContext {
+    public:
+        virtual ~UserContext() {};
```

Comment:
It makes sense to put destructor implementation in .cpp file.
Otherwise "full header" classes may lead to RTTI issues (e.g. with clang - it creates independent clones in each module)